tools/rosbag/src/record.cpp

```diff
@@ -141,16 +141,15 @@ rosbag::RecorderOptions parseOptions(int argc, char** argv) {
         }
         double freq = it->second.as<double>();
         opts.topics.push_back(it->first.as<std::string>());
-        if (freq == -1)
+
+        if (freq == 0)
         {
-          continue;
-        }
-        if (freq < 0.001 && freq > 0)
+            opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration(-1));
+        } 
+        else 
         {
-            ROS_WARN("Topic has freq less than 0.001, Topic name:= %s ; Topic will be recorded at 0.001 hz", it->first.as<std::string>().c_str());
-            freq = 0.001;
+            opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration(double(1)/freq));
```

Comment:
just 
`opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration(freq > 0 ? 1/freq : -1));`
would be more elegant. Also might want to set any negative number to -1 due to check below? Or change check below from `== ros::Duration(-1)` to `< ros::Duration(0)` ?

Reply:
it can be taken care in the next regex PR